mjml/__init__.pyi

```diff
@@ -0,0 +1,35 @@
+import pathlib
+import typing as t
+
+if t.TYPE_CHECKING:
+    import typing_extensions as te
+    from _typeshed import SupportsRead
+
+    from mjml.core.api import Component
+
+    class _Output(t.NamedTuple):
+        html: str
+        errors: t.Sequence[str]
+
+        @te.overload
+        def __getitem__(self, _: t.Literal["html"]) -> str: ...
+        @te.overload
+        def __getitem__(self, _: t.Literal["errors"]) -> t.Sequence[str]: ...
+        @te.overload
+        def get(self, key: t.Literal["html"], /) -> t.Optional[str]: ...
+        @te.overload
+        def get(self, key: t.Literal["html"], default: str, /) -> str: ...
+        @te.overload
+        def get(self, key: t.Literal["errors"], /) -> t.Optional[t.Sequence[str]]: ...
+        @te.overload
+        def get(self, key: t.Literal["errors"], default: t.Sequence[str], /) -> t.Sequence[str]: ...
+
+    FpOrJson = t.Union[t.Dict[str, t.Any], str, bytes, SupportsRead[str], SupportsRead[bytes]]
+    StrOrPath = t.Union[str, pathlib.PurePath]
+
+def mjml_to_html(
+    xml_fp_or_json: "FpOrJson",
+    skeleton: t.Optional[str] = None,
+    template_dir: t.Optional["StrOrPath"] = None,
```

Comment:
This is one of the places where types are really helpful as I looked through the code and I am not sure if a `str` instance is really acceptable. It seems to me at first glance that we are doing `template_dir / foo` unconditionally so this should be `os.PathLike`.

Maybe you can check my finding? (I am probably overlooking something)

Reply:
> It seems to me at first glance that we are doing `template_dir / foo` unconditionally so this should be `os.PathLike`.

This should be fine, as long as `foo` is a `pathlib.PurePath` (or derivative thereof), and it looks to me like this is in fact the case.

I used this annotation initially because there's already some string usage (as far as I can tell), for instance here:
https://github.com/FelixSchwarz/mjml-python/blob/ed3187e2d7f22ab39848f056e7406ef0b2118aff/mjml/scripts/mjml.py#L32